hail/src/main/scala/is/hail/variant/ReferenceGenome.scala

```diff
@@ -271,7 +269,7 @@ case class ReferenceGenome(name: String, contigs: Array[String], lengths: Map[St
       }
     }
 
-    if (!Interval.isValid(locusType.ordering, start, end, includesStart, includesEnd))
+    if (!Interval.isValid(locusType.ordering(HailStateManager(Map(this.name -> this))), start, end, includesStart, includesEnd))
```

Comment:
don't love having to allocate a bunch of stuff here for this check (currently it's creating the ordering too, but once the Type.scala change is made that will return the cached ordering). How about we make `mkOrdering(rg)` / `ordering(rg)` methods on TLocus mirroring mkOrdering(sm) and ordering(sm), which `mkOrdering(sm)` calls through to after looking up the rg?

Reply:
Hm, I agree and it feels silly that we're constructing a map with one element just to take that one element out of it. But also a little funny that we're calling locusType.ordering(rg) that will just turn around and call rg.locusOrdering. I tried just moving the implementation over to ReferenceGenome so that instead of having 1 indirection in TLocus and 2 indirections in ReferenceGenome we have no indirections in ReferenceGenome and still 1 indirection in TLocus.

Reply:
oh, yeah, great.